test/test_functional.py

```diff
@@ -1,90 +1,78 @@
 import torch
 from torchtext import functional
+from .common.parameterized_utils import nested_params
 
 from .common.torchtext_test_case import TorchtextTestCase
 
-
 class TestFunctional(TorchtextTestCase):
-    def _to_tensor(self, test_scripting):
-        input = [[1, 2], [1, 2, 3]]
-        padding_value = 0
 
+    @nested_params(
+        [True, False],
+        [
+            [[[1, 2], [1, 2, 3]], 0, [[1, 2, 0], [1, 2, 3]]],
+            [[[1, 2], [1, 2, 3]], 1, [[1, 2, 1], [1, 2, 3]]],
+            [[1, 2], 0, [1, 2]],
+        ]
+    )
+    def test_to_tensor(self, test_scripting, configs):
+        """test tensorization on both single sequence and batch of sequence"""
+        inputs, padding_value, expected_list = configs
         func = functional.to_tensor
         if test_scripting:
             func = torch.jit.script(func)
-        actual = func(input, padding_value=padding_value)
-        expected = torch.tensor([[1, 2, 0], [1, 2, 3]], dtype=torch.long)
-        torch.testing.assert_close(actual, expected)
+        else:
+            with self.assertRaises(TypeError):
+                func("test")
 
-        input = [1, 2]
-        actual = func(input, padding_value=padding_value)
-        expected = torch.tensor([1, 2], dtype=torch.long)
+        actual = func(inputs, padding_value=padding_value)
+        expected = torch.tensor(expected_list, dtype=torch.long)
         torch.testing.assert_close(actual, expected)
 
-    def test_to_tensor(self):
-        """test tensorization on both single sequence and batch of sequence"""
-        self._to_tensor(test_scripting=False)
-
-    def test_to_tensor_jit(self):
-        """test tensorization with scripting on both single sequence and batch of sequence"""
-        self._to_tensor(test_scripting=True)
-
-    def _truncate(self, test_scripting):
+    @nested_params(
+        [True, False],
+        [
+            [[[1, 2], [1, 2, 3]], [[1, 2], [1, 2]]],
+            [[1, 2, 3], [1, 2]],
+            [[["a", "b"], ["a", "b", "c"]], [["a", "b"], ["a", "b"]]],
+            [["a", "b", "c"], ["a", "b"]],
+        ]
+    )
+    def test_truncate(self, test_scripting, configs):
+        """test truncation on both sequence and batch of sequence with both str and int types"""
```

Comment:
This docstring does not explain what the actual testing is. Could you rephrase in a way that better explains what the passing criteria is? such as something should match with somethig

Reply:
Done. See #1667.